lib/application.js

```diff
@@ -158,6 +158,11 @@ module.exports = class Application extends Emitter {
       keys: this.keys,
       secure: request.secure
     });
+    if (req && req.headers) {
+      request.ip = request.ips[0] || req.socket.remoteAddress || '';
+    } else {
+      request.ip = ''; // We are in a badly written test :P
```

Comment:
I'm not sure if we should throw here(don't know what's the impact when can't get IP), but at lease we should give more detailed information here rather than just an assert.